persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java

```diff
@@ -91,15 +121,22 @@ private WriteOptions dataWriteOptions() {
         return dataWriteOptions;
     }
 
-    private Options dataDbOptions() {
-        Options options = new Options().setCreateIfMissing(true);
-        options.setCompressionType(CompressionType.getCompressionType(configuration.compressionType().toString()));
+    private DBOptions dataDbOptions() {
+        DBOptions options = new DBOptions()
```

Comment:
if you could use DBOptions.getDBOptionsFromProps(configuration.properties()) we should be able to take care of https://issues.jboss.org/browse/ISPN-9371

Reply:
The only problem with that is now we have `DBOptions` and `ColumnFamilyOptions`. I would have to check if it ignores improper values and subsequently determine if we should set compression or not.

Reply:
I think this needs to be thought through a little bit more in a different PR. Unfortunately now that we are using column families they allow for different properties than `DBOptions`. And both of those are slightly different than the `Options` class.

`Options` - used when a database is only configured with the default column family (aka no column families)
`DBOptions` - used when column families are used and includes config for things not covered by column families
`ColumnFamilyOptions` - used for each column family things such as compression etc.

Reply:
Also to keep in mind you can't provide a non mapped property, otherwise RocksDB will return null options object back. So we need a way to namespace these. I was thinking maybe something like prefix names `database` and `data` for `DBOptions` and `ColumnFamilyOptions` respectively.

Reply:
Yes. And fail fast if it the null dboptions is returned